File: Tema 05 - Typescript/PetShelterFrontend/src/models/Fund.ts
import type { Person } from "./Person";

export class Fund {
    name: string;
    donationAmount: number;
    goal: number;
    creationDate: Date;
    dueDate: Date;
    status: string;
    owner?: Person;

    constructor(name: string, donationAmount: number, goal: number, creationDate: Date, dueDate: Date, status: string, owner?: Person) {
        this.name = name;
        this.donationAmount = donationAmount;
        this.goal = goal;
        this.creationDate = creationDate;
        this.dueDate = dueDate;
        this.status = status;
        this.owner = owner;
    }
}

File: Tema 07 - Clean Code/After/petshelter.api/PetShelter.DataAccessLayer/Models/Fund.cs
namespace PetShelter.DataAccessLayer.Models
{
    public class Fund : IEntity
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public decimal TotalDonationAmount { get; set; }
        public decimal Goal { get; set; }
        public DateTime CreationDate { get; set; } = DateTime.Now;
        public DateTime DueDate { get; set; }
        public string Status { get; set; }
        public int? OwnerId { get; set; }

        public Person Owner { get; set; }

        public ICollection<Donation> FundDonations { get; set; }
    }
}
